handler.go

```diff
@@ -60,8 +62,22 @@ func (o Options) ReadOptions(key string, obj interface{}) error {
 	return json.Unmarshal(msg, obj)
 }
 
+// GenesisParams represents parameters set in genesis that could be useful
+// for some of the extensions.
+type GenesisParams struct {
+	Validators []abci.ValidatorUpdate
+}
+
+// FromInitChain initialises GenesisParams using abci.RequestInitChain
+// data.
+func FromInitChain(req abci.RequestInitChain) GenesisParams {
```

Comment:
I try to keep all the abci-type stuff in `app` and just the internal weave types in this top-level dir. Not a hard rule, but nice if possible to move it.

Reply:
you mean referencing here: ```	Validators []abci.ValidatorUpdate```

Yeah, we can move it.

Reply:
Hmm.. I guess this is abci types all the way down...

Anyway, do what makes most sense, as long as the use of the api is clear looking at the weave code without having to dive into the tendermint codebase, I am fine. And it would be a huge headache to remove all abci references, so do what makes sense. The above was more an opinion than a requirement